rio_tiler/io/base.py

```diff
@@ -48,12 +48,9 @@ class SpatialMixin:
     height: Optional[int] = attr.ib(default=None, init=False)
     width: Optional[int] = attr.ib(default=None, init=False)
 
-    geographic_crs: CRS = attr.ib(init=False, default=WGS84_CRS)
-
-    @cached_property
-    def geographic_bounds(self) -> BBox:
-        """Return dataset bounds in geographic_crs."""
-        if self.crs == self.geographic_crs:
+    def get_geographic_bounds(self, crs: CRS) -> BBox:
```

Comment:
`get_geographic_bounds` or `geographic_bounds`?

Reply:
on balance `get_geographic_bounds` as it's no longer a cached property, but it's just a preference and not a huge concern to me